packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1933,12 +2029,22 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 
   @override
   void didChangeNext(Route<dynamic>? nextRoute) {
+    if (nextRoute is ModalRoute<T> && canTransitionTo(nextRoute) && nextRoute.delegatedTransition != this.delegatedTransition) {
```

Comment:
This may be come an issue if you add multiple route through page api.

e.g.
https://gist.github.com/chunhtai/5396b38c5584a8badc3b875db78fe65b

Reply:
no, jump to middle is a bug, I filed https://github.com/flutter/flutter/issues/156033

as for correct behavior, I think it would be as if one the top most page pushed onto the home page with override.

let me see if I can put up a quick fix for the jumping to middle problem

Reply:
actually I think we are fine. after fixing the bug, the animation looks good

Reply:
So this is isn't a blocker to merge this PR?

Reply:
no, it is not. we are good